modmail/extensions/utils/error_handler.py

```diff
@@ -0,0 +1,195 @@
+                        f"Attempting to dm {ctx.author} since they have a permission which may be able "
+                        "to give the bot send message permissions."
+                    )
+                    try:
+                        await ctx.author.send(embeds=[embed])
+                    except discord.Forbidden:
+                        logger.notice("Also encountered an error when trying to reply in dms.")
+                        return False
+            return True
+
+    async def handle_check_failure(
+        self, ctx: commands.Context, error: commands.CheckFailure
+    ) -> typing.Optional[discord.Embed]:
+        """Handle CheckFailures seperately given that there are many of them."""
+        title = "Check Failure"
+        msg = None
+        if isinstance(error, commands.CheckAnyFailure):
+            title = self.get_title_from_name(error.checks[-1])
+            msg = str(error)
+        elif isinstance(error, commands.PrivateMessageOnly):
+            title = "DMs Only"
+        elif isinstance(error, commands.NoPrivateMessage):
+            title = "Server Only"
+        elif isinstance(error, commands.BotMissingPermissions):
+            # defer handling BotMissingPermissions to a method, since this can be problematic
+            await self.handle_bot_missing_perms(ctx, error)
+            return None
+        else:
+            title = self.get_title_from_name(error)
+        embed = self.error_embed(msg or str(error), title=title)
+        return embed
+
+    @ModmailCog.listener()
+    async def on_command_error(self, ctx: commands.Context, error: commands.CommandError) -> None:
+        """Activates when a command raises an error."""
+        if getattr(error, "handled", False):
+            logging.debug(f"Command {ctx.command} had its error already handled locally; ignoring.")
+            return
+
+        if isinstance(error, commands.CommandNotFound):
+            # ignore every time the user inputs a message that starts with our prefix but isn't a command
+            # this will be modified in the future to support prefilled commands
+            if ANY_DEV_MODE:
+                logger.trace(error)
+            return
+
+        logger.trace(error)
+
+        title = None
+        msg = None
+        embed: typing.Optional[discord.Embed] = None
+        should_respond = True
+
+        if isinstance(error, commands.UserInputError):
+            embed = await self.handle_user_input_error(ctx, error)
+        elif isinstance(error, commands.CheckFailure):
+            embed = await self.handle_check_failure(ctx, error)
+            # handle_check_failure may send its own error if its a BotMissingPermissions error.
+            if embed is None:
+                should_respond = False
+        elif isinstance(error, commands.ConversionError):
+            # s = object()
+            # error.converter.convert.__annotations__.get("return", s)
+            # embed = error
+            ...
```

Comment:
Incomplete? @onerandomusername
If we actually want incomplete fallthrough, a 
`pass  # fall through to default response formatting` would be the best way to annotate it I think.

Reply:
Yeah, I think I meant to do something but can remove this.